llvm/lib/CodeGen/GCEmptyBasicBlocks.cpp

```diff
@@ -26,6 +27,17 @@ using namespace llvm;
 
 STATISTIC(NumEmptyBlocksRemoved, "Number of empty blocks removed");
 
+static bool removeEmptyBlocks(MachineFunction &MF);
+
+PreservedAnalyses
+GCEmptyBasicBlocksPass::run(MachineFunction &MF,
+                            MachineFunctionAnalysisManager &MFAM) {
+  bool Changed = removeEmptyBlocks(MF);
+  if (Changed)
+    return getMachineFunctionPassPreservedAnalyses();
+  return PreservedAnalyses::all();
+}
+
 class GCEmptyBasicBlocks : public MachineFunctionPass {
```

Comment:
I would recommend renaming the class name `GCEmptyBasicBlocks` to `GCEmptyBasicBlocksLegacy` to indicate the legacy pass definition.